### PR TITLE
Quick-fix for the Settings navigation overflow issue

### LINKED ISSUE #7938: App icon disappears for overflowing settings page names
### Steps to reproduce
1. Create app with a rather long settings page name
2. Apply app-icon to app
3. App icon will not be shown in e.g. /index.php/settings/user

### Expected behaviour
Settings page name gets shortened, but app icon stays visible.

### Actual behaviour
Settings page name is shortened and app icon disappears.

### Nextcloud version
13.0.0RC1

bildschirmfoto 2018-01-18 um 14 58 46

This is an issue that probably needs refactoring of the HTML structure.

We have this
```
<a href="/nextcloud/index.php/settings/admin/my_app">
  <img alt="" src="/nextcloud/apps/my_app/img/app-dark.svg">
  <span>A very long app settings page name that is too long</span>
</a>
```

And because the a-tag gets a padding at the position where the img is placed, it will be hidden, once the span-text overflows, because of overflow: hidden. At this moment I have no simple idea how to solve this, maybe someone else has.

### PR DESCRIPTION
**Pull request by pixelipo**
It's ugly and I don't particularly like it, but it's the only simple way to fix #7938

### CODE REVIEW

**Comment by te-online** [comment]
Thanks for the fix! Unfortunately, I can only review this on Monday.

File: core/css/apps.scss [hunk 1]
```diff
@@ -524,6 +530,16 @@ kbd {
 	}
 }
 
+/* patch to fix overflow in the Settings page */
```
Comment (juliushaertl) [inline core/css/apps.scss]: I'd say we should move this to settings/css/settings.scss then.

Comment (skjnldsv) [reply core/css/apps.scss]: Not since it can also be used in other apps. :/

Comment (juliushaertl) [reply core/css/apps.scss]: Ok, but in that case I would not separate it form the general rules. :wink:

**Comment by skjnldsv** [comment]
@pixelipo could you put the flex into the global popovermenu rule? It should not change for links not using img inside I thing.

We need to check with some example tough, like with menu, with counter, with both, with submenu... etc :)

**Comment by pixelipo** [comment]
What do you mean popovermenu, @skjnldsv ? This is the `#app-navigation` bug, not popover...

**Comment by skjnldsv** [comment]
@pixelipo sorry, I mean the app navigation! I was working on popover and had that in mind :)

**Comment by pixelipo** [comment]
@skjnldsv I've tried that, but then it breaks overflow in all the other places. It also requires a fix for the quota element. I'll push a commit now so you can have a look

**Comment by skjnldsv** [comment]
Indeed. :/
Maybe we should stop trying to support img in the navigation and only use the css classes like we decided to go before? Having that many trouble to implement something like that isn't really worth it imho. img are supposed to be an element itself. The icons in the app-navigation are decorative, they're not meant to have a sizeable value. Having them as background was the goal, so they're not selectionable nor part of the dom.

**Comment by jancborchardt** [comment]
@pixelipo @te-online @skjnldsv any update here? :) We’re in the middle of the 15 cycle so perfect time to get it finished and merged.

**Comment by pixelipo** [comment]
Proper way to do this would be to replace `` with icon classes in the Settings app.

I can't commit to that kind of work right now. Maybe in a couple of weeks.

**Comment by pixelipo** [comment]
News is using it in a way that wouldn't be broken by this, it adds `style='background-image: url("")` to have favicons, but default is icon class.

**Comment by te-online** [comment]
@pixelipo @jancborchardt Can you reproduce the original issue https://github.com/nextcloud/server/issues/7938 on `master`. I cannot even reproduce on the latest stable release... Maybe it just solved itself? 😉 

bildschirmfoto 2018-10-05 um 23 31 23

**Comment by skjnldsv** [comment]
> @skjnldsv @juliushaertl what do you think? Going with CSS instead of img tags seems good, as we already do that lots of places. Question is if it also works for cases where favicons or custom icons are used, like in the News app (in case we do standardize it).

We'll need a fake icon-custom (for example) to tell the css there is an icon and not an img, but yes it's perfect.
Like pixelipo said, this will works fine as it's already standardised :)

**Comment by te-online** [comment]
@pixelipo Strangely, I cannot reproduce on `master`. Maybe some kind of CSS cache going on? (I cleared browser cache and `data/appdata_*/css`.

Also, with your changes, everything seems fine, but how can I make any reliable statement about that when I can't even reproduce the original issue...

bildschirmfoto 2018-10-08 um 22 02 28

**Comment by jancborchardt** [comment]
So how do we continue here? :) @te-online you reported the original issue https://github.com/nextcloud/server/issues/7938 but can’t reproduce it anymore, correct?

@pixelipo what do you think? And @nextcloud/designers?

**Pull request closed without merging**